src/chainstate/burn/db/burndb.rs

```diff
@@ -421,363 +439,150 @@ const BURNDB_SETUP : &'static [&'static str]= &[
-        rng.fill_bytes(&mut buf);
-        let db_path_dir = format!("/tmp/test-blockstack-burndb-{}", to_hex(&buf));
-        let (db_path, index_path) = db_mkdirs(&db_path_dir)?;
-
-        debug!("Open {}", &db_path);
-        let mut conn = Connection::open_with_flags(&db_path, OpenFlags::SQLITE_OPEN_READ_WRITE | OpenFlags::SQLITE_OPEN_CREATE).map_err(db_error::SqliteError)?;
-        conn.busy_handler(Some(tx_busy_handler)).map_err(db_error::SqliteError)?;
-
-        BurnDB::instantiate(&mut conn, &index_path, first_block_height, first_burn_hash, get_epoch_time_secs())?;
-
-        let marf = BurnDB::open_index(&index_path)?;
-
-        let db = BurnDB {
-            conn: conn,
-            marf: marf,
-            readwrite: true,
-            first_block_height: first_block_height,
-            first_burn_header_hash: first_burn_hash.clone(),
-        };
-        
-        Ok(db)
+fn get_ancestor_sort_id<C: SortitionContext>(ic: &IndexDBConn<'_, C, SortitionId>, block_height: u64, tip_block_hash: &SortitionId) -> Result<Option<SortitionId>, db_error> {
+    let first_block_height = ic.context.first_block_height();
+    if block_height < first_block_height {
+        return Ok(None);
     }
+ 
+    ic.get_ancestor_block_hash(block_height - first_block_height, &tip_block_hash)
+}
 
-    /// Open the database on disk.  It must already exist and be instantiated.
-    /// It's best not to call this if you are able to call connect().  If you must call this, do so
-    /// after you call connect() somewhere else, since connect() performs additional validations.
-    pub fn open(path: &str, readwrite: bool) -> Result<BurnDB, db_error> {
-        let open_flags =
-            if readwrite {
-                OpenFlags::SQLITE_OPEN_READ_WRITE
-            }
-            else {
-                OpenFlags::SQLITE_OPEN_READ_ONLY
-            };
-
-        let (db_path, index_path) = db_mkdirs(path)?;
-        debug!("Open burndb '{}' as '{}', with index as '{}'",
-               db_path, if readwrite { "readwrite" } else { "readonly" }, index_path);
-        
-        let conn = Connection::open_with_flags(&db_path, open_flags).map_err(db_error::SqliteError)?;
-        conn.busy_handler(Some(tx_busy_handler)).map_err(db_error::SqliteError)?;
-
-        let marf = BurnDB::open_index(&index_path)?;
-        let first_snapshot = BurnDB::get_first_block_snapshot(&conn)?;
+pub struct SortitionId(pub [u8; 32]);
+impl_array_newtype!(SortitionId, u8, 32);
+impl_array_hexstring_fmt!(SortitionId);
+impl_byte_array_newtype!(SortitionId, u8, 32);
+impl_byte_array_from_column!(SortitionId);
+impl_byte_array_message_codec!(SortitionId, 32);
 
-        let db = BurnDB {
-            conn: conn,
-            marf: marf,
-            readwrite: readwrite,
-            first_block_height: first_snapshot.block_height,
-            first_burn_header_hash: first_snapshot.burn_header_hash.clone(),
-        };
-        Ok(db)
-    }
+pub struct PoxIdentifier(pub [u8; 32]);
```

Comment:
For keeping the codebase consistent, can this struct's name be changed to `PoxId`?